templates/pkg/resource/manager.go.tpl

```diff
@@ -156,6 +162,69 @@ func (rm *resourceManager) ARNFromName(name string) string {
 	)
 }
 
+// LateInitialize returns an acktypes.AWSResource after setting the late initialized
+// fields from the readOne call. This method will initialize the optional fields
+// which were not provided by the k8s user but were defaulted by the AWS service.
+// If there are no such fields to be initialized, the returned object is similar to
+// object passed in the parameter.
+func (rm *resourceManager) LateInitialize(
+	ctx context.Context,
+	res acktypes.AWSResource,
+) (acktypes.AWSResource, error) {
+	rlog := ackrtlog.FromContext(ctx)
+{{- if $hookCode := Hook .CRD "late_initialize_pre_read_one" }}
+{{ $hookCode }}
+{{- end }}
+	// If there are no fields to late initialize, do nothing
+	if len(lateInitializeFieldNames) == 0 {
+		rlog.Debug("no late initialization required.")
+		return res, nil
+	}
+	r := rm.concreteResource(res)
+	if r.ko == nil {
+		// Should never happen... if it does, it's buggy code.
+		panic("resource manager's LateInitialize() method received resource with nil CR object")
+	}
+	koWithDefaults := r.ko.DeepCopy()
+	latestWithDefaults := &resource{koWithDefaults}
+	lateInitConditionReason := ""
+	lateInitConditionMessage := ""
+
+	observed, err := rm.sdkFind(ctx, latestWithDefaults)
```

Comment:
Instead of calling `sdkFind` with a `*resource`, you can just call `ReadOne` and pass `res`, like so:

```go
var latest acktypes.AWSResource
if latest, err := rm.ReadOne(ctx, res); err != nil {
    	lateInitConditionMessage = "Unable to complete Read operation required for late initialization"
	lateInitConditionReason = "Late Initialization Failure"
	ackcondition.SetLateInitialized(
		latest,
		corev1.ConditionFalse,
		&lateInitConditionMessage,
		&lateInitConditionReason,
	)
	return latest, ackrequeue.NeededAfter(err, time.Duration(0)*time.Second)
}
```

Reply:
Thanks. I will make the change.